processor/src/trace/mod.rs

```diff
@@ -134,6 +134,10 @@ impl ExecutionTrace {
         result
     }
 
+    pub fn get_trace_len(&self) -> usize {
+        self.main_trace.num_rows()
+    }
+
     // HELPER METHODS
```

Comment:
This is weird that I had to create a new public function. I tried to use `trace.length()` from https://github.com/maticnetwork/miden/blob/aadd94187f724186b0df08c57f3b87dd7ec5b308/processor/src/trace/mod.rs#L166 but I couldn't do it properly.

Reply:
I think the issue there is because we implement this method as trait, and the trait isn't public. Any binary must use only public API. The solution would be either move `Trace` to the public scope, or leave a getter (as you did). I prefer the latter; also, moving `Trace` to public would be outside the scope of this PR

Reply:
Good so leave as is?

Reply:
I would have preferred to expose it as just `length()`. We usually use `get_` prefix on methods which are not simple accessors. For example, we could do it like this:
```Rust
pub fn get_trace_len(&self) -> usize {
    <Self as Trace>::length(self)
}
```

Reply:
Actually, I think we've even been preferring `len` over `length` in most places?

Reply:
It seems this prints the number of rows in the trace after the padding to a power of two, can we get the number of rows prior to that?

Reply:
Yep - I've left a comment about this [here](https://github.com/0xPolygonMiden/miden-vm/issues/442#issuecomment-1372754958).

Reply:
Will work on this hopefully this Friday